test/common/QRCodeStructuredAppendReader.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+* Copyright 2016 Nu-book Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "QRCodeStructuredAppendReader.h"
+#include "HybridBinarizer.h"
+#include "TextUtfEncoding.h"
+#include "Result.h"
+#include "DecodeHints.h"
+#include "qrcode/QRReader.h"
+#include "ImageLoader.h"
+
+#include <algorithm>
+
+namespace ZXing { namespace Test {
+
+QRCodeStructuredAppendReader::QRCodeStructuredAppendReader(const std::shared_ptr<ImageLoader>& imgLoader)
+: _imageLoader(imgLoader)
+{
+}
+
+TestReader::ReadResult
+QRCodeStructuredAppendReader::readMultiple(const std::vector<std::wstring>& filenames, int rotation) const
+{
+	TestReader::ReadResult result;
+	DecodeHints hints;
+	QRCode::Reader reader(hints);
+	std::list<Result> allResults;
+	int prevParity = 0;
```

Comment:
I have no idea how that parity feature works either, but since it seems to behave like an 8-bit checksum that needs to be equal in all code symbols (judged by looking at your code), I suggest to change the 'uninitialized value' to `-1` to make sure it can not be a value that is actually returned from `getBits()`.